File: 1253-reconstruct-a-2-row-binary-matrix/1253-reconstruct-a-2-row-binary-matrix.java
class Solution {
    public List<List<Integer>> reconstructMatrix(int upper, int lower, int[] colsum) {
        int n = colsum.length;
        List<List<Integer>> result = new ArrayList<>();
        Integer upperRow[] = new Integer[n];
        Integer lowerRow[] = new Integer[n];
        Arrays.fill(upperRow, 0);
        Arrays.fill(lowerRow, 0);
        for (int i = 0; i < n; i++) {
            if (colsum[i] == 2) {
                upperRow[i] = 1;
                lowerRow[i] = 1;
                upper--;
                lower--;
            }
        }
        for (int i = 0; i < n; i++) {
            if (colsum[i] == 1) {
                if (upper > 0) {
                    upperRow[i] = 1;
                    upper--;
                } else if (lower > 0) {
                    lowerRow[i] = 1;
                    lower--;
                } else {
                    return result;
                }
            }
        }
        if (upper != 0 || lower != 0) {
            return result;
        }
        result.add(Arrays.asList(upperRow));
        result.add(Arrays.asList(lowerRow));
        return result;
    }
}

File: 1253-reconstruct-a-2-row-binary-matrix/1253-reconstruct-a-2-row-binary-matrix.py
class Solution:
    def reconstructMatrix(self, upper: int, lower: int, colsum: List[int]) -> List[List[int]]:
        n = len(colsum)
        upper_list = [0 for _ in range(n)]
        lower_list = [0 for _ in range(n)]
        
        for i, v in enumerate(colsum):
            if v == 1:
                if upper > lower:
                    upper_list[i] = 1
                    upper -= 1
                else:    
                    lower_list[i] = 1
                    lower -= 1
            elif v == 2:        
                upper_list[i] = lower_list[i] = 1
                upper, lower = upper - 1, lower - 1
                
        return [upper_list, lower_list] if upper == lower == 0 else []